### ISSUE TITLE
[FR]: Create a way to ignore checked in files

### ISSUE THREAD

**Issue by mrmeku** [aspect-build/rules_lint#151]
### What is the current behavior?

Currently, the only way to ignore files is for use with the prettier extension since it is a binary that has its own ignoring logic using `.prettierignore` or other means particular to that binary

For other languages, such as go, there is no way to not have a file be formattted. This poses many problems for checked in generated code, vendored in libraries and other situations in which formatting is simply not desired for particular use cases. The only lever available is to turn off formatting for the entire language itself

We cannot rely on each formatter that bazel-super-formatter uses to have its own ignore logic. 

### Describe the feature

We need to introduce a file which bazel-super-formatter can parse for `.gitignore` like patterns for exclusion. The file can be a standardized location (such as .prettierignore at the root of the repo) but generally, it would be better us to expose a parameter that is configurable in a bazelrc file.

`run --@aspect_rules_format//ignore="path/to/file"`

In general, I believe the best way to faciliate this is to replace the file identification step of [format.sh](https://github.com/aspect-build/bazel-super-formatter/blob/main/format/format.sh)  with a binary

IE replace
```
if [ "$#" -eq 0 ]; then
  files=$(git ls-files '*.py' '*.pyi')
else
  files=$(find "$@" -name '*.py' -or -name '*.pyi')
fi
```
with
```
bin=$(rlocation aspect_rules_format/find_files_binary)
files=$(bin '*.py' '*.pyi')
```

As to which language this binary should be written in, that's up for debate. It would be nice to do this only in bash, but I doubt that there is a bash utility installed on all systems that has this use case built in. That being the case, it seems logical
to use a language whose toolchain we already install for formatting. This leaves us with Go or Javascript as primary candidates.

### Fund our work

- [ ] Sponsor our open source work by donating a [feature bounty](https://opencollective.com/aspect-build/)

**Comment by ewhauser** [comment]
One way to do this is to make your formatter and sh_binary and do additional filtering in a bash script before calling the underlying formatter tool.

We use this to use two different SQL formatters for different needs (one for Postgres and one for dbt)

**Comment by alexeagle** [comment]
I think the README section "Ignoring files" answers this, though we could discuss whether something is really absent that's making it hard for users.

**Comment by alexeagle** [comment]
I guess we need to do this. Philosophy is "do the same thing you would do outside of Bazel" -  I imagine ppl already have some place they had to list these generated files, e.g. we already have https://github.com/aspect-build/rules_lint/blob/main/.gitattributes
OR maybe `.editorconfig` or similar.

**Comment by alexeagle** [comment]
Fixed by #210
